### PR TITLE
[4.x] MFA Backup Codes [a11y] [UI]

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #43449 .

### Summary of Changes
1. Combine nested if
2. add a class to the link to force it to be underlined

### Testing Instructions
Enable any of the MFA methods and then try to print out the backup codes

### Actual result BEFORE applying this Pull Request
chrome_3Ns56XJJ8c

### Expected result AFTER applying this Pull Request
image

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by brianteeman** [comment]
This is an option. Another would be to change it from a string and link to a button similar to 
image

**Comment by richard67** [comment]
@brianteeman Does this close the referred issue #43449 ?

**Comment by brianteeman** [comment]
it's an option

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on c43291024edf9fbfd2b60e6a83af17addbfccf53<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43450">issues.joomla.org/tracker/joomla-cms/43450</a>.</sub>

File: components/com_users/tmpl/methods/list.php [hunk 1]
```diff
@@ -69,12 +69,10 @@ class="img-fluid"
...
-                                    <?php if ($methodName === 'backupcodes') : ?>
-                                        <?php if ($canAddEdit) : ?>
-                                            <div class="alert alert-info mt-1 w-100">
-                                                <?php echo Text::sprintf('COM_USERS_MFA_BACKUPCODES_PRINT_PROMPT_HEAD', Route::_('index.php?option=com_users&task=method.edit&id=' . (int) $record->id . ($this->returnURL ? '&returnurl=' . $this->escape(urlencode($this->returnURL)) : '') . '&user_id=' . $this->user->id)) ?>
-                                            </div>
-                                        <?php endif ?>
+                                    <?php if ($methodName === 'backupcodes' && $canAddEdit) : ?>
+                                        <div class="alert alert-info mt-1 w-100">
+                                            <?php echo Text::sprintf('COM_USERS_MFA_BACKUPCODES_PRINT_PROMPT_HEAD', Route::_('index.php?option=com_users&task=method.edit&id=' . (int) $record->id . ($this->returnURL ? '&returnurl=' . $this->escape(urlencode($this->returnURL)) : '') . '&user_id=' . $this->user->id)) ?>
```
Comment (Quy) [inline components/com_users/tmpl/methods/list.php]:
```suggestion
                                            <?php echo Text::sprintf('COM_USERS_MFA_BACKUPCODES_PRINT_PROMPT_HEAD', Route::_('index.php?option=com_users&task=method.edit&id=' . (int) $record->id . ($this->returnURL ? '&returnurl=' . $this->escape(urlencode($this->returnURL)) : '') . '&user_id=' . $this->user->id), 'text-decoration-underline') ?>
```

File: administrator/components/com_users/tmpl/methods/list.php [hunk 2]
```diff
@@ -69,12 +69,10 @@ class="img-fluid"
...
-                                        <?php if ($canAddEdit) : ?>
-                                            <div class="alert alert-info mt-1 w-100">
-                                                <?php echo Text::sprintf('COM_USERS_MFA_BACKUPCODES_PRINT_PROMPT_HEAD', Route::_('index.php?option=com_users&task=method.edit&id=' . (int) $record->id . ($this->returnURL ? '&returnurl=' . $this->escape(urlencode($this->returnURL)) : '') . '&user_id=' . $this->user->id)) ?>
-                                            </div>
-                                        <?php endif ?>
+                                    <?php if ($methodName === 'backupcodes' && $canAddEdit) : ?>
+                                        <div class="alert alert-info mt-1 w-100">
+                                            <?php echo Text::sprintf('COM_USERS_MFA_BACKUPCODES_PRINT_PROMPT_HEAD', Route::_('index.php?option=com_users&task=method.edit&id=' . (int) $record->id . ($this->returnURL ? '&returnurl=' . $this->escape(urlencode($this->returnURL)) : '') . '&user_id=' . $this->user->id), 'text-decoration-underline') ?>
```
Comment (Quy) [inline administrator/components/com_users/tmpl/methods/list.php:74]: Maybe use `alert-link`?

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on 559d520b520f975dd7c8c881cd8c56111a2f3f9a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43450">issues.joomla.org/tracker/joomla-cms/43450</a>.</sub>

File: language/en-GB/com_users.ini [hunk 3]
```diff
@@ -45,7 +45,7 @@ COM_USERS_MAIL_SEND_FAILURE_SUBJECT="Error sending email"
 COM_USERS_MFA_ADD_AUTHENTICATOR_OF_TYPE="Add a new %s"
 COM_USERS_MFA_ADD_PAGE_HEAD="Add a Multi-factor Authentication Method"
 COM_USERS_MFA_BACKUPCODES_PRINT_PROMPT="Backup Codes let you log into the site if your regular Multi-factor Authentication method does not work or you no longer have access to it. Each code can be used <em>only once</em>."
-COM_USERS_MFA_BACKUPCODES_PRINT_PROMPT_HEAD="<a href=\"%s\">Print these codes</a> and keep them in your wallet."
+COM_USERS_MFA_BACKUPCODES_PRINT_PROMPT_HEAD="<a href=\"%1s\" class=\"%2s\">Print these codes</a> and keep them in your wallet."
```
Comment (Quy) [inline language/en-GB/com_users.ini:48]: It is already underlined on the frontend. I guess it ok to add the class anyway for consistency.

Comment (brianteeman) [reply language/en-GB/com_users.ini:48]: accessibility!! not consistency ;)

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 559d520b520f975dd7c8c881cd8c56111a2f3f9a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43450">issues.joomla.org/tracker/joomla-cms/43450</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43450">issues.joomla.org/tracker/joomla-cms/43450</a>.</sub>

File: language/en-GB/com_users.ini (hunk 3 above)
Comment (laoneo) [reply language/en-GB/com_users.ini:48]: @Quy when ok, please resolve the conversation so we can merge it.

**Comment by MacJoom** [comment]
Thank you!

**Pull request merged**